app/services/RestaurantService.java

```diff
@@ -172,61 +172,44 @@ public Restaurant getRestaurantWithId(final UUID id)
      * @return the popular restaurants
      */
     @SuppressWarnings("unchecked")
-    public List<Restaurant> getPopularRestaurants(UUID userId)
+    public PaginationAdapter<Restaurant> getPopularRestaurants(User user)
     {
+	if (user != null)
+	{
 
-	UserHabitsBean userHabits = getUserHabits(userId);
-      
-	List<Restaurant> restaurants = (List<Restaurant>) getSession().createCriteria(Restaurant.class)
-		.add(Restrictions.eq("city.id", userHabits.getCity()))
-		.add(Restrictions.eq("priceRange", userHabits.getUserPriceRange())).list();
+	    UserHabitsBean userHabits = getUserHabits(user);
+	    if (userHabits != null)
+	    {
 
-	List<Object> toRemove = new ArrayList<Object>();
+		List<Restaurant> restaurants = (List<Restaurant>) getSession().createCriteria(Restaurant.class)
+			.add(Restrictions.eq("city.id", userHabits.getCity()))
+			.add(Restrictions.eq("priceRange", userHabits.getUserPriceRange())).list();
+	
 
-	for (Restaurant restaurant : restaurants)
-	{
+		List<Restaurant> toRemove = new ArrayList<Restaurant>();
 
-	    if (restaurant.getAverageRating() < (userHabits.getUserRatingRange() - 0.5d)
-		    || restaurant.getAverageRating() > (userHabits.getUserRatingRange() + 0.5d))
-	    {
-		toRemove.add(restaurant);
-	    }
+		for (Restaurant restaurant : restaurants)
+		{
 
-	}
-	restaurants.removeAll(toRemove);
+		    if (restaurant.getAverageRating() < (userHabits.getUserRatingRange() - 0.5d)
+			    || restaurant.getAverageRating() > (userHabits.getUserRatingRange() + 0.5d))
+		    {
+			toRemove.add(restaurant);
+		    }
 
-	if (restaurants.size() > 0)
-	{
-	    return restaurants;
-	}
+		}
+		restaurants.removeAll(toRemove);
```

Comment:
This is still not calculated inside the actual query

Reply:
Nisam to uradio, cini mi se da nema referenca koju si pominjao u prethodnom komitu